### PR TITLE
Add Nevergrad optimization of the introduced (non-differentiable) metrics

### LINKED ISSUE #7: Nevergrad optimizers
Currently we only have a random optimizer. This is related to the fact that our scores (Spearman's rho, MI, Kendall's tau) are non-differentiable. We can use [nevergrad](https://facebookresearch.github.io/nevergrad/) for this.

### PR DESCRIPTION
**Pull request by pawel-czyz**
This PR resolves #7 by adding Nevergrad optimization. The interfaces between Nevergrad and random sampling are as unified as possible.
The tests are now split into separate files and disentangled – they run faster and should be easier to maintain (or extend).

Current limitations (to be addressed in future PRs; they do not seem urgent at the moment):
- Saving optimization history (and making it work with batch-based optimization and multithreading);
- Allowing additional callbacks (as logging).

### CODE REVIEW

File: tests/test_direction.py [hunk 1]
```diff
@@ -49,8 +49,8 @@ def test_find_gradient(true_direction: Sequence[float], metric: di.AxisMetric) -
     scores = score_according_to_gradient(direction=true_direction, points=data)
 
     # Retrieve the vector yielding best scores
-    optimization_result = di.random.find_best_direction(
-        points=data, scores=scores, metric=metric, budget=500, random_generator=5
+    optimization_result = di.optimization.find_best_direction(
```
Comment (AnejSvete) [inline tests/test_direction.py:52]: So we do not test the random module anymore?

File: latte/direction/random.py [hunk 2]
```diff
@@ -106,7 +111,5 @@ def find_best_direction(
     budget: int = 200,
     random_generator: Union[int, np.random.Generator] = 1,
```
Comment (AnejSvete) [inline latte/direction/random.py:112]: Should we also rename this parameter or no?

**Review by AnejSvete — COMMENTED** [review]
Looks good to me, just the two small questions. :)

File: latte/direction/random.py (hunk 2 above)
Comment (pawel-czyz) [reply latte/direction/random.py]: Good catch! Renamed :slightly_smiling_face:

File: latte/direction/random.py [hunk 3]
```diff
@@ -1,10 +1,15 @@
...
+For more sophisticated approaches, see `latte.direction.optimization`.
+"""
 import dataclasses
 from typing import NamedTuple, Sequence, Union
 
 import numpy as np
 import numpy.typing as ntypes
 
-import latte.direction.common as common
+import latte.direction.metrics as common
```
Comment (AnejSvete) [inline latte/direction/random.py:12]: Should we also rename the alias?

**Review by AnejSvete — APPROVED** [review]
Perfect :+1:.

**Pull request merged**